pkg/azure/azureClientWrapper/azureClientWrapper.go

```diff
@@ -70,9 +73,9 @@ func NewAzureClientWrapper(logger *slog.Logger, subscriptionId string, credentia
 	}, nil
 }
 
-var ErrPageAdvanceFailure = errors.New("failed to advance page")
-
 func (a *AzClientWrapper) ListVirtualMachineScaleSetsOwnedVms(ctx context.Context, rgName, vmssName string) ([]*armcompute.VirtualMachineScaleSetVM, error) {
+	listLogger := a.logger.With("pager", "listVirtualMachineScaleSetsOwnedVms")
```

Comment:
Just a nit, what's your reasoning for `listLogger`? I think `logger := ...` would be clear enough.

Reply:
no reason... just (as jjo says) E_NAMING_IS_HARD :)

I'll change to `logger`, it makes more sense.